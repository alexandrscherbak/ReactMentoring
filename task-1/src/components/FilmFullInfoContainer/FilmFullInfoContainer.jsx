import React from 'react';
import Logo from '../Logo/Logo.jsx';
import FilmImage from '../FilmInfoContainer/FilmImage/FilmImage.jsx';

export default class FilmFullInfoContainer extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div className="film-full-info-container">
				<div className="film-full-info-header-row">
					<Logo />
					<button
						onClick={() => this.props.searchClicked()}
					>Search</button>
				</div>
				<div className="film-full-info">
					<div className="film-image">
						<img src={this.props.filmInfo.poster_path} alt={this.props.filmInfo.title} />
					</div>
					<div className="film-info">
						<h1>{this.props.filmInfo.title}</h1>
						<div className="release-and-rating">
							<span>{this.props.filmInfo.release_date.slice(0, 4)}</span>
							<span>{this.props.filmInfo.vote_average}</span>
						</div>
						<span>{this.props.filmInfo.overview}</span>
					</div>
				</div>

				<style jsx>
					{`
						.film-full-info-container {
							padding: 20px 100px;
							background-image: url('../headerBackground2.jpg');
						}
						.film-image {
							flex-basis: 300px;
							flex-shrink: 0;
						}
						img {
							max-width: 100%;
						}
						.film-full-info-header-row {
							display: flex;
							justify-content: space-between;
						}
						h1 {
							color: pink;
						}
						.film-full-info {
							display: flex;
						}
						.film-info {
							margin-left: 50px;
						}
						.release-and-rating {
							display: flex;
							justify-content: space-between;
						}
					`}
				</style> 
			</div>
		);
	}
};
