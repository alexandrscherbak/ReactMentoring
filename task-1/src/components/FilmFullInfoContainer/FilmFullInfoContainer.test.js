import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmFullInfoContainer from './FilmFullInfoContainer.jsx';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const props = {
	filmInfo: {
		poster_path: 'poster_path',
		title: 'film',
		release_date: '2018-04-01',
		vote_average: '7',
		overview: 'overview',
	},
	searchClicked: jest.fn(),
};

function setup() {
	return mount(<FilmFullInfoContainer {...props} />);
}

describe('FilmFullInfoContainer', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<FilmFullInfoContainer {...props}/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render self and subcomponents', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.find('img').html()).toContain(props.filmInfo.title);
		expect(enzymeWrapper.find('.film-info').childAt(0).text()).toBe(props.filmInfo.title);
		expect(enzymeWrapper.find('.film-info').childAt(1).childAt(0).text()).toBe(props.filmInfo.release_date.slice(0, 4));
		expect(enzymeWrapper.find('.film-info').childAt(1).childAt(1).text()).toBe(props.filmInfo.vote_average);
		expect(enzymeWrapper.find('.film-info').childAt(2).text()).toBe(props.filmInfo.overview);
	});
	it('should call searchClicked if button clicked', () => {
		const enzymeWrapper = setup();
		const button = enzymeWrapper.find('button');
		expect(props.searchClicked.mock.calls.length).toBe(0);
		button.props().onClick();
		expect(props.searchClicked.mock.calls.length).toBe(1);
	});
});
