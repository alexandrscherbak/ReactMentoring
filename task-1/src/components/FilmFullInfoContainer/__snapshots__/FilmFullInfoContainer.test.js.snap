// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmFullInfoContainer snapshot 1`] = `
<div
  className="jsx-1209675933 film-full-info-container"
>
  <div
    className="jsx-1209675933 film-full-info-header-row"
  >
    <div
      className="jsx-2852891300"
    >
      <span
        className="jsx-2852891300"
      >
        netflixroulette
      </span>
    </div>
    <button
      className="jsx-1209675933"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <div
    className="jsx-1209675933 film-full-info"
  >
    <div
      className="jsx-1209675933 film-image"
    >
      <img
        alt="film"
        className="jsx-1209675933"
        src="poster_path"
      />
    </div>
    <div
      className="jsx-1209675933 film-info"
    >
      <h1
        className="jsx-1209675933"
      >
        film
      </h1>
      <div
        className="jsx-1209675933 release-and-rating"
      >
        <span
          className="jsx-1209675933"
        >
          2018
        </span>
        <span
          className="jsx-1209675933"
        >
          7
        </span>
      </div>
      <span
        className="jsx-1209675933"
      >
        overview
      </span>
    </div>
  </div>
</div>
`;
